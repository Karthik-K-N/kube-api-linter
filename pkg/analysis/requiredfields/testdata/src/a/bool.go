package a

type TestBools struct {
	// +kubebuilder:validation:Required
	Bool bool `json:"bool"` // want "field Bool should have the omitempty tag." "field Bool has a valid zero value \\(false\\) and should be a pointer."

	// +kubebuilder:validation:Required
	BoolWithOmitEmpty bool `json:"boolWithOmitEmpty,omitempty"` // want "field BoolWithOmitEmpty has a valid zero value \\(false\\) and should be a pointer."

	// +kubebuilder:validation:Required
	BoolPtr *bool `json:"boolPtr"` // want "field BoolPtr should have the omitempty tag."

	// +kubebuilder:validation:Required
	BoolPtrWithOmitEmpty *bool `json:"boolPtrWithOmitEmpty,omitempty"`
}
