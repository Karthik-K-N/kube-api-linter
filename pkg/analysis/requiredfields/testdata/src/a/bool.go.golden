package a

type TestBools struct {
    // +kubebuilder:validation:Required
	Bool *bool `json:"bool,omitempty"` // want "field Bool should have the omitempty tag." "field Bool has a valid zero value \\(false\\) and should be a pointer."

    // +kubebuilder:validation:Required
	BoolWithOmitEmpty *bool `json:"boolWithOmitEmpty,omitempty"` // want "field BoolWithOmitEmpty has a valid zero value \\(false\\) and should be a pointer."

    // +kubebuilder:validation:Required
	BoolPtr *bool `json:"boolPtr,omitempty"` // want "field BoolPtr should have the omitempty tag."

    // +kubebuilder:validation:Required
	BoolPtrWithOmitEmpty *bool `json:"boolPtrWithOmitEmpty,omitempty"`
}
